kusto/kusto.go

```diff
@@ -133,7 +145,7 @@ func New(endpoint string, auth Authorization, options ...Option) (*Client, error
 		)
 	}
 
-	client := &Client{auth: auth, endpoint: endpoint}
```

Comment:
Again this change from auth to *auth is possibly going to be questioned. Is there a specific reason for this